dbt/adapters/snowflake/relation_configs/dynamic_table.py

```diff
@@ -21,6 +35,147 @@ class SnowflakeDynamicTableConfig(RelationConfigBase):
     """
 
     name: str
+    schema_name: str
+    database_name: str
     query: str
     target_lag: SnowflakeDynamicTableTargetLagConfig
     warehouse: str
+
+    @property
+    def path(self) -> str:
+        return ".".join(
+            part for part in [self.database_name, self.schema_name, self.name] if part is not None
```

Comment:
So what happens if `self.schema_name` is `None` but the other's aren't?

Reply:
lol, I thought the same thing, but this is how we roll in `dbt-core`. fwiw, those three attributes are required on this class, but could be provided as `""`. I'm fine dealing with that edge case since the user would effectively be choosing to break it. That becomes more of an issue when this winds up in a base class in `dbt-core` (down the road). If you're still concerned, I would address it with a validation rule, checking that all three parts are of non-zero length (rather than complicate this logic).

Reply:
I don't love this but I'll take the explanation.